rasterio/_crs.pyx

```diff
@@ -0,0 +1,56 @@
+"""Coordinate reference systems, class and functions.
+"""
+
+import logging
+
+from rasterio cimport _gdal, _base
+from rasterio.compat import UserDict
+from rasterio.compat import string_types
+
+log = logging.getLogger(__name__)
+
+class _CRS(UserDict):
+    """
+    """
+    @property
+    def is_geographic(self):
+        cdef void *osr_crs = _base._osr_from_crs(self)
+        cdef int retval
+        try:
+            retval = _gdal.OSRIsGeographic(osr_crs)
+            return bool(retval == 1)
+        finally:
+            _gdal.OSRDestroySpatialReference(osr_crs)
+
+    @property
+    def is_projected(self):
+        cdef void *osr_crs = _base._osr_from_crs(self)
+        cdef int retval
+        try:
+            retval = _gdal.OSRIsProjected(osr_crs)
+            return bool(retval == 1)
+        finally:
+            _gdal.OSRDestroySpatialReference(osr_crs)
+
+    def __eq__(self, other):
+        cdef void *osr_crs1 = _base._osr_from_crs(self)
```

Comment:
@jameshiebert this is another place where `try` / `finally` might be appropriate too.

For the above `try` / `finally` they are probably OK as written, but I think my preference would be (untested code):
```
cdef void *osr_crs = NULL
cdef int retval
try:
    osr_crs = _base._osr_from_crs(self)
    retval = _gdal.OSRIsGeographic(osr_crs)
    return bool(retval == 1)
finally:
    _gdal.OSRDestroySpatialReference(osr_crs)
```

Just to avoid any possibility of leaving that `osr_crs` lying around.